code/modules/power/lighting.dm

```diff
@@ -23,6 +23,19 @@ TYPEINFO(/obj/item/light_parts)
 	var/light_type = /obj/item/light/tube
 	var/fitting = "tube"
 	var/install_type = INSTALL_WALL
+	var/image/light_image = null
+
+	New()
+		..()
+		update_light_tube_overlay()
+
+	proc/update_light_tube_overlay()
```

Comment:
Could likely be override of `update_icon` (which is called by the wrapper `UpdateIcon`). This is useful in case something external modifies us etc.

Also `light_image` does not need to be a var on the object. It's ok to make it a local variable here or even better use `src.SafeGetOverlayImage("light")`. That proc will reuse the last image used for this overlay or create a new one if there is no such overlay.

Reply:
Yeah this makes much more sense, ty pushed these changes